Library/Homebrew/hardware.rb

```diff
@@ -152,26 +163,32 @@ def arm?
         type == :arm
       end
 
+      sig { returns(T::Boolean) }
       def little_endian?
         !big_endian?
       end
 
+      sig { returns(T::Boolean) }
       def big_endian?
         [1].pack("I") == [1].pack("N")
       end
 
+      sig { returns(FalseClass) }
       def virtualized?
         false
       end
 
+      sig { returns(T::Array[String]) }
```

Comment:
```suggestion
      sig { returns(T::Array[Symbol]) }
```

the extend/os equivalents seem to be symbols